src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -24,63 +25,67 @@
 import java.util.*;
 import java.util.stream.Collectors;
 
+/**
+ * This class describes a colony and contains all the data and methods for manipulating a Colony.
+ */
 public class Colony implements IColony
 {
-    private         final   int                             id;
+    private final int id;
 
     //  Runtime Data
-    private                 World                           world                           = null;
+    private World world = null;
 
     //  Updates and Subscriptions
-    private                 Set<EntityPlayerMP>             subscribers                     = new HashSet<>();
-    private                 boolean                         isDirty                         = false;
-    private                 boolean                         isCitizensDirty                 = false;
-    private                 boolean                         isBuildingsDirty                = false;
+    private Set<EntityPlayerMP> subscribers = new HashSet<>();
+    private boolean isDirty = false;
+    private boolean isCitizensDirty = false;
+    private boolean isBuildingsDirty = false;
+    private boolean manualHiring = false;
 
     //  General Attributes
-    private                 String                          name                            = "ERROR(Wasn't placed by player)";
-    private         final   int                             dimensionId;
-    private                 BlockPos                        center;
+    private String name = "ERROR(Wasn't placed by player)";
+    private final int dimensionId;
+    private BlockPos center;
 
     //  Administration/permissions
-    private                 Permissions                     permissions                     = new Permissions();
+    private Permissions permissions = new Permissions();
     //private int autoHostile = 0;//Off
 
     //  Buildings
-    private BuildingTownHall                                townHall;
-    private                 Map<BlockPos, Building>         buildings                       = new HashMap<>();
+    private BuildingTownHall townHall;
+    private Map<BlockPos, AbstractBuilding> buildings = new HashMap<>();
 
     //  Citizenry
-    private                 Map<Integer, CitizenData>       citizens                        = new HashMap<>();
-    private                 int                             topCitizenId                    = 0;
-    private                 int                             maxCitizens                     = Configurations.maxCitizens;
-
+    private Map<Integer, CitizenData> citizens = new HashMap<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private field 'citizens' could be made final; it is only initialized in the declaration or constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AImmutableField)